project_close_forecast_date/models/purchase_order_line.py

```diff
@@ -0,0 +1,20 @@
+# © 2024 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import _, api, fields, models
+from pytz import timezone
+
+
+class PurchaseOrderLine(models.Model):
+    _inherit = 'purchase.order.line'
+
+    forecast_date = fields.Date(compute='_compute_forecast_date', store=True, readonly=True)
+
+    @api.depends('date_planned') 

```

Comment:
If this dependency is added, write() stuff for stock.picking should become unneeded? Please check (when picking is created and when picking is modified):
```suggestion
    @api.depends('date_planned', 'move_ids.picking_id.scheduled_date') 
```